src/content/docs/en/reference/container-reference.mdx

```diff
@@ -0,0 +1,292 @@
+### Rendering options
+
+Both [`renderToResponse`](#rendertoresponse) and [`renderToString`](#rendertostring) accept an object as their second argument:
+
+```ts
+export type ContainerRenderOptions = {
+	slots?: Record<string, any>;
+	request?: Request;
+	params?: Record<string, string | undefined>;
+	locals?: App.Locals;
+	routeType?: "page" | "endpoint";
+  
+};
+```
+
+#### `slots`
+
+**Type**: `Record<string, any>`;
+
+Use this options if your component needs to render some [slots](en/basics/astro-components/#slots).
+
+If your components renders one slot, pass an object with `default` as key:
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  slots: { default: "Some value"}
+});
+```
+
+If your component renders named slots, use the keys of the object to name the slots:
+
+```astro name="Card.astro"
+---
+---
+<div>
+  <slot name="header" />
+  <slot name="footer" />
+</div>
+```
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  slots: { "header": "Header content", "footer": "Footer" }
+});
+```
+
+You can also render components in cascade:
+
+```astro name="Card.astro"
+---
+---
+<div>
+  <slot name="header" />
+  <slot name="footer" />
+</div>
+```
+
+```js name="Card.test.js"
+import Card from "../src/components/Card.astro";
+import CardHeader from "../src/components/CardHeader.astro";
+import CardFooter from "../src/components/CardFooter.astro";
+
+const result = await container.renderToString(Card, { 
+  slots: { 
+    "header": await container.renderToString(CardHeader), 
+    "footer":  await container.renderToString(CardFooter),
+  }
+});
+```
+
+#### `request` option
+
+**Type**: `Request`
+
+The request is used to understand which path/URL the component is about to render.
+
+Use this option in case your component needs to read information like `Astro.url` or `Astro.request`.
+
+You can also inject possible headers or cookies.
+
+```js file="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  request: new Request("https://example.com/blog", {
+    headers: {
+      "X-some-secret-header": "test-value"
+    }
+  })
+});
+```
+
+#### `params` option
+
+**Type**: `Record<string, string | undefined>`;
+
+Use this option in case your component needs to read `Astro.params`. As opposed to `getStaticPaths`, you must provide only **one** item of the params returned by that function.
+
+```astro name="Card.astro"
+---
+const { locale, slug } = Astro.params;
+---
+<div></div>
+```
+
+```js file="Card.test.js"
+import Card from "../src/components/Card.astro";
+
+const result = await container.renderToString(Card, { 
+  params: {
+    locale: "en",
+    slug: "getting-started"
+  }
+});
+```
+
+#### `locals` options 
+
+**Type**: `App.Locals`
+
+Use this option to stub the `Astro.locals` object:
```

Comment:
```suggestion
An option to pass information from [`Astro.locals`](/en/reference/api-reference/#astrolocals) for rendering your component.

Use this option to when your component needs information stored during the lifecycle of a request in order to render, such as logged in status.
```
These words might be kind of wrong, but just trying to fit this into the pattern!